configuration/src/main/java/io/camunda/configuration/beanoverrides/CamundaExporterConfigurationApplier.java

```diff
@@ -0,0 +1,238 @@
+            });
+  }
+
+  public static void applyBulk(
+      final ExporterConfiguration exporterConfiguration,
+      final UnifiedConfiguration unifiedConfiguration) {
+
+    getDocumentBasedDatabase(unifiedConfiguration)
+        .ifPresent(
+            source -> {
+              final BulkConfiguration target = exporterConfiguration.getBulk();
+              target.setDelay(Math.toIntExact(source.getBulk().getDelay().getSeconds()));
+              target.setSize(source.getBulk().getSize());
+              target.setMemoryLimit(
+                  Math.toIntExact(source.getBulk().getMemoryLimit().toMegabytes()));
+            });
+  }
+
+  public static void applyIncidentNotifier(
+      final ExporterConfiguration exporterConfiguration,
+      final UnifiedConfiguration unifiedConfiguration) {
+
+    getDocumentBasedDatabase(unifiedConfiguration)
+        .ifPresent(
+            source -> {
+              final IncidentNotifierConfiguration notifier = exporterConfiguration.getNotifier();
+              notifier.setWebhook(source.getIncidentNotifier().getWebhook());
+              notifier.setAuth0Domain(source.getIncidentNotifier().getAuth0Domain());
+              notifier.setAuth0Protocol(source.getIncidentNotifier().getAuth0Protocol());
+              notifier.setM2mClientId(source.getIncidentNotifier().getM2mClientId());
+              notifier.setM2mClientSecret(source.getIncidentNotifier().getM2mClientSecret());
+              notifier.setM2mAudience(source.getIncidentNotifier().getM2mAudience());
+            });
+  }
+
+  public static void applyMisc(
+      final ExporterConfiguration exporterConfiguration,
+      final UnifiedConfiguration unifiedConfiguration) {
+
+    getDocumentBasedDatabase(unifiedConfiguration)
+        .ifPresent(
+            source -> {
+              exporterConfiguration.setCreateSchema(source.isCreateSchema());
+              exporterConfiguration
+                  .getBatchOperationCache()
+                  .setMaxCacheSize(source.getBatchOperationCache().getMaxSize());
+              exporterConfiguration
+                  .getBatchOperation()
+                  .setExportItemsOnCreation(source.getBatchOperations().isExportItemsOnCreation());
+              exporterConfiguration
+                  .getProcessCache()
+                  .setMaxCacheSize(source.getProcessCache().getMaxSize());
+              exporterConfiguration
+                  .getDecisionRequirementsCache()
+                  .setMaxCacheSize(source.getDecisionRequirementsCache().getMaxSize());
+              exporterConfiguration
+                  .getFormCache()
+                  .setMaxCacheSize(source.getFormCache().getMaxSize());
+            });
+
+    exporterConfiguration.setAuditLog(
+        unifiedConfiguration.getCamunda().getData().getAuditLog().toConfiguration());
+  }
+
+  private static Optional<DocumentBasedSecondaryStorageDatabase> getDocumentBasedDatabase(
```

Comment:
🔧 I think this does not have to be an Optional, right? We can just assume it's there, as the validation happens before calling this code. Worst case would be a NullPointerException, which is reasonable as something is completely off with the config code (as mentioned, its should have been validated before).

Not making this an Optional also simplifies the calling code (in all of the other methods).

We could change this to something like

```java
return unifiedConfiguration
        .getCamunda()
        .getData()
        .getSecondaryStorage()
        .getDocumentBasedDatabase()
        .orElseThrow(...);
```